packages/core/test/actor.test.ts

```diff
@@ -1105,4 +1105,111 @@ describe('actors', () => {
       })
       .start();
   });
+
+  // https://github.com/statelyai/xstate/issues/2507
+  it('should not crash on sync child completion before stopping', () => {
```

Comment:
I ain't sure what the "before stopping" part is referring to here, it might be that I'm reading this in a wrong way 🤔  Maybe something like:
```suggestion
  it('should not crash on sync child completion during self-initialization', () => {
```

nit: all added tests are actually testing the same thing, just with different data types (machines, promises, observables etc) but yet their respective test titles are different. Could we unify them? It would get more clear that they are related - you could maybe even create a `describe` block for all of them